bloptools/bayesian/__init__.py

```diff
@@ -288,13 +247,153 @@ def tell(self, new_table=None, append=True, train=True, **kwargs):
             try:
                 self.train_models()
             except botorch.exceptions.errors.ModelFittingError:
-                if self._initialized:
+                if self.initialized:
                     self._set_hypers(cached_hypers)
                 else:
                     raise RuntimeError("Could not fit model on initialization!")
 
         self.constraint = GenericDeterministicModel(f=lambda x: self.classifier.probabilities(x)[..., -1].squeeze(-1))
 
+    def ask(self, acq_func_identifier="qei", n=1, route=True, sequential=True, **acq_func_kwargs):
+        """
+        Ask the agent for the best point to sample, given an acquisition function.
+
+        acq_func_identifier: which acquisition function to use
+        n: how many points to get
```

Comment:
Let's make it `numpydoc`-compatible: https://numpydoc.readthedocs.io/en/latest/format.html#parameters.